Ontime.js

```diff
@@ -303,52 +309,47 @@ function changeEvent(
   title,
   note,
   cue,
+  timeStart,
+  linkStart,
+  timeEnd,
   duration,
   skip,
   public,
+  timerType,
+  endAction,
+  timeWarning,
+  timeDanger,
   selectColour,
   pickColour,
   writeColour,
   customID,
-  customText,
+  customText
 ) {
-  if (action == 'title') {
-    local.send('{"type":"change", "payload":{"' + id + '":{"title":"' + title + '"}}}');
-  } else if (action == 'note') {
-    local.send('{"type":"change", "payload":{"' + id + '":{"note":"' + note + '"}}}');
-  } else if (action == 'cue') {
-    local.send('{"type":"change", "payload":{"' + id + '":{"cue":"' + cue + '"}}}');
-  } else if (action == 'duration') {
-    duration = parseInt(duration);
-    local.send('{"type":"change", "payload":{"' + id + '":{"duration":' + duration + '}}}');
-  } else if (action == 'skip') {
-    local.send('{"type":"change", "payload":{"' + id + '":{"' + action + '":' + skip + '}}}');
-  } else if (action == 'isPublic') {
-    local.send('{"type":"change", "payload":{"' + id + '":{"isPublic":' + public + '}}}');
-  } else if (action == 'selectColour') {
-    //user selects enumerated value.
-    if (color == 'none') {
-      local.send('{"type":"change", "payload":{"' + id + '":{"colour":""}}}');
-    } else {
-      local.send('{"type":"change", "payload":{"' + id + '":{"colour":"' + selectColour + '"}}}'); //using enum
-    }
-  } else if (action == 'pickColour') {
-    script.log('color');
-    var hexColor =
-      '#' +
+  // TODO: a timeout that won't let users spam change requests?
+  var changeAction = {
+    'title': ['title', title],
+    'note': ['note', note],
+    'cue': ['cue',cue],
+    'timeStart': ['timeStart',parseInt(timeStart)],
+    'linkStart': ['linkStart',linkStart],
+    'timeEnd': ['timeEnd',parseInt(timeEnd)],
+    'duration': ['duration',parseInt(duration)],
+    'skip': ['skip',skip],
+    'isPublic': ['isPublic',public],
+    'timerType': ['timerType',timerType], // Find correct values for function?
+    'endAction': ['endAction',endAction], // Find correct values for function?
```

Comment:
> Find correct values for function?

What are you missing here?

Reply:
Running these as is, results in Error: Invalid value received

Reply:
Oh, ok will look into that as well 
It looks correct from your end